test-workflow.yml

```diff
@@ -0,0 +1,83 @@
+# Clubhouse Task (#645): 
+# Create a Github Action (GHA) to run our tests every time a PR is opened (or a branch is updated within a PR). 
+# This is called Continuous Integration (CI) -- the idea being that we are continually testing our code.
+
+
+# Questions: 
+# 1. Which tests should be included? All? 
+# 2. Should I run the tests with pytest?
+# 3. Should this be based off the notebooks (such as the dataset publishing notebook)? 
+# 4. Should I use Flake8? ("uses it to lint all files") <- idk what that means
+
+name: test-workflow
```

Comment:
@imogen-foster were you able to run this before PRing?  I believe as it's written, it should work on every push (unless I'm mistaken). you can run things on the `imogen-foster-gha-workflow-tests-1` branch without doing a PR into `main`. you can edit the `on: [push]` block to work with pushing to `imogen-foster-gha-workflow-tests-1` specifically, too